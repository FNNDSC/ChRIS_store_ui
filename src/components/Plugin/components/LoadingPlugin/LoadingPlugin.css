.loading-plugin-container {
  /* display */
  display: flex;
  flex-direction: column;
}

.loading-plugin-container, .plugin-body {
  /* flex */
  flex: 1;
}

.loading-plugin-tag {
  /* display */
  display: inline-block;
}