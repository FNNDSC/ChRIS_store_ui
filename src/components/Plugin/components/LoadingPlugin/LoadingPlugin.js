import React from 'react';
import LoadingContainer from '../../../LoadingContainer/LoadingContainer';
import LoadingContent from '../../../LoadingContainer/components/LoadingContent/LoadingContent';
import './LoadingPlugin.css';

const LoadingPlugin = () => (
  <LoadingContainer className="loading-plugin-container">
    <div className="plugin-header">
      <div className="row no-flex">
        <LoadingContent
          width="8em"
          height="1.5em"
        />
        <LoadingContent
          width="12em"
          height="2.3em"
          top="1em"
        />
        <LoadingContent
          width="13em"
          height="1em"
          top="0.8em"
        />
        <LoadingContent
          className="loading-plugin-tag"
          width="4em"
          height="1.5em"
          top="0.5em"
        />
        <LoadingContent
          className="loading-plugin-tag"
          width="14em"
          height="1.5em"
          top="1em"
          left="0.5em"
        />
      </div>
    </div>
    <div className="plugin-body">
      <div className="row">
        <div className="plugin-body-main-col">
          <LoadingContent
            type="white"
            width="100%"
            height="9em"
          />
          <LoadingContent
            type="white"
            width="100%"
            height="9em"
            top="1em"
          />
        </div>
        <div className="plugin-body-side-col">
          <LoadingContent
            type="white"
            width="100%"
            height="9em"
          />
        </div>
      </div>
    </div>
  </LoadingContainer>
);

export default LoadingPlugin;
