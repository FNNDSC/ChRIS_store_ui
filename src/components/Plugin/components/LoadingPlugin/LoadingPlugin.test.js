import React from 'react';
import { shallow } from 'enzyme';
import LoadingPlugin from './LoadingPlugin';

describe('LoadingPlugin', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LoadingPlugin />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render LoadingContainer component', () => {
    expect(wrapper.find('LoadingContainer.loading-plugin-container')).toHaveLength(1);
  });

  /* ============================== */
  /* ====== PLUGIN HEADER DIV ===== */
  /* ============================== */

  it('should render plugin-header div inside LoadingContainer', () => {
    expect(wrapper
      .find('LoadingContainer')
      .find('div.plugin-header'))
      .toHaveLength(1);
  });

  it('should render no-flex row div inside plugin-header', () => {
    expect(wrapper
      .find('div.plugin-header')
      .find('div.row.no-flex'))
      .toHaveLength(1);
  });

  it('should render 5 LoadingContent components inside plugin-header', () => {
    expect(wrapper
      .find('div.plugin-header')
      .find('LoadingContent'))
      .toHaveLength(5);
  });

  /* ============================== */
  /* ======= PLUGIN BODY DIV ====== */
  /* ============================== */

  it('should render plugin-body div inside LoadingContainer', () => {
    expect(wrapper
      .find('LoadingContainer')
      .find('div.plugin-body'))
      .toHaveLength(1);
  });

  it('should render row div inside plugin-body', () => {
    expect(wrapper
      .find('div.plugin-body')
      .find('div.row'))
      .toHaveLength(1);
  });

  it('should render 3 LoadingContent components in plugin-body with type="white', () => {
    wrapper
      .find('div.plugin-body')
      .find('LoadingContent')
      .forEach((loadingContent) => {
        expect(loadingContent.is('[type="white"]')).toBeTruthy();
      });
  });

  it('should render plugin-body-main-col div inside plugin-body', () => {
    expect(wrapper
      .find('div.plugin-body')
      .find('div.plugin-body-main-col'))
      .toHaveLength(1);
  });

  it('should render plugin-body-side-col div inside plugin-body', () => {
    expect(wrapper
      .find('div.plugin-body')
      .find('div.plugin-body-side-col'))
      .toHaveLength(1);
  });

  it('should render 2 LoadingContent components in plugin-body-main-col', () => {
    expect(wrapper
      .find('div.plugin-body')
      .find('div.plugin-body-main-col')
      .find('LoadingContent'))
      .toHaveLength(2);
  });

  it('should render 1 LoadingContent component in plugin-body-side-col', () => {
    expect(wrapper
      .find('div.plugin-body')
      .find('div.plugin-body-side-col')
      .find('LoadingContent'))
      .toHaveLength(1);
  });
});
