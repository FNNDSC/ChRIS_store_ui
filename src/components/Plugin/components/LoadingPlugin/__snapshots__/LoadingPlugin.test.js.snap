// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingPlugin should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingPlugin />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="plugin-header"
>
          <div
                    className="row no-flex"
          >
                    <LoadingContent
                              bottom="0"
                              className=""
                              height="1.5em"
                              left="0"
                              right="0"
                              top="0"
                              type=""
                              width="8em"
                    />
                    <LoadingContent
                              bottom="0"
                              className=""
                              height="2.3em"
                              left="0"
                              right="0"
                              top="1em"
                              type=""
                              width="12em"
                    />
                    <LoadingContent
                              bottom="0"
                              className=""
                              height="1em"
                              left="0"
                              right="0"
                              top="0.8em"
                              type=""
                              width="13em"
                    />
                    <LoadingContent
                              bottom="0"
                              className="loading-plugin-tag"
                              height="1.5em"
                              left="0"
                              right="0"
                              top="0.5em"
                              type=""
                              width="4em"
                    />
                    <LoadingContent
                              bottom="0"
                              className="loading-plugin-tag"
                              height="1.5em"
                              left="0.5em"
                              right="0"
                              top="1em"
                              type=""
                              width="14em"
                    />
          </div>
</div>,
        <div
          className="plugin-body"
>
          <div
                    className="row"
          >
                    <div
                              className="plugin-body-main-col"
                    >
                              <LoadingContent
                                        bottom="0"
                                        className=""
                                        height="9em"
                                        left="0"
                                        right="0"
                                        top="0"
                                        type="white"
                                        width="100%"
                              />
                              <LoadingContent
                                        bottom="0"
                                        className=""
                                        height="9em"
                                        left="0"
                                        right="0"
                                        top="1em"
                                        type="white"
                                        width="100%"
                              />
                    </div>
                    <div
                              className="plugin-body-side-col"
                    >
                              <LoadingContent
                                        bottom="0"
                                        className=""
                                        height="9em"
                                        left="0"
                                        right="0"
                                        top="0"
                                        type="white"
                                        width="100%"
                              />
                    </div>
          </div>
</div>,
      ],
      "className": "loading-plugin-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row no-flex"
>
            <LoadingContent
                        bottom="0"
                        className=""
                        height="1.5em"
                        left="0"
                        right="0"
                        top="0"
                        type=""
                        width="8em"
            />
            <LoadingContent
                        bottom="0"
                        className=""
                        height="2.3em"
                        left="0"
                        right="0"
                        top="1em"
                        type=""
                        width="12em"
            />
            <LoadingContent
                        bottom="0"
                        className=""
                        height="1em"
                        left="0"
                        right="0"
                        top="0.8em"
                        type=""
                        width="13em"
            />
            <LoadingContent
                        bottom="0"
                        className="loading-plugin-tag"
                        height="1.5em"
                        left="0"
                        right="0"
                        top="0.5em"
                        type=""
                        width="4em"
            />
            <LoadingContent
                        bottom="0"
                        className="loading-plugin-tag"
                        height="1.5em"
                        left="0.5em"
                        right="0"
                        top="1em"
                        type=""
                        width="14em"
            />
</div>,
          "className": "plugin-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <LoadingContent
                bottom="0"
                className=""
                height="1.5em"
                left="0"
                right="0"
                top="0"
                type=""
                width="8em"
/>,
              <LoadingContent
                bottom="0"
                className=""
                height="2.3em"
                left="0"
                right="0"
                top="1em"
                type=""
                width="12em"
/>,
              <LoadingContent
                bottom="0"
                className=""
                height="1em"
                left="0"
                right="0"
                top="0.8em"
                type=""
                width="13em"
/>,
              <LoadingContent
                bottom="0"
                className="loading-plugin-tag"
                height="1.5em"
                left="0"
                right="0"
                top="0.5em"
                type=""
                width="4em"
/>,
              <LoadingContent
                bottom="0"
                className="loading-plugin-tag"
                height="1.5em"
                left="0.5em"
                right="0"
                top="1em"
                type=""
                width="14em"
/>,
            ],
            "className": "row no-flex",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "bottom": "0",
                "className": "",
                "height": "1.5em",
                "left": "0",
                "right": "0",
                "top": "0",
                "type": "",
                "width": "8em",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "bottom": "0",
                "className": "",
                "height": "2.3em",
                "left": "0",
                "right": "0",
                "top": "1em",
                "type": "",
                "width": "12em",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "bottom": "0",
                "className": "",
                "height": "1em",
                "left": "0",
                "right": "0",
                "top": "0.8em",
                "type": "",
                "width": "13em",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "bottom": "0",
                "className": "loading-plugin-tag",
                "height": "1.5em",
                "left": "0",
                "right": "0",
                "top": "0.5em",
                "type": "",
                "width": "4em",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "bottom": "0",
                "className": "loading-plugin-tag",
                "height": "1.5em",
                "left": "0.5em",
                "right": "0",
                "top": "1em",
                "type": "",
                "width": "14em",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
>
            <div
                        className="plugin-body-main-col"
            >
                        <LoadingContent
                                    bottom="0"
                                    className=""
                                    height="9em"
                                    left="0"
                                    right="0"
                                    top="0"
                                    type="white"
                                    width="100%"
                        />
                        <LoadingContent
                                    bottom="0"
                                    className=""
                                    height="9em"
                                    left="0"
                                    right="0"
                                    top="1em"
                                    type="white"
                                    width="100%"
                        />
            </div>
            <div
                        className="plugin-body-side-col"
            >
                        <LoadingContent
                                    bottom="0"
                                    className=""
                                    height="9em"
                                    left="0"
                                    right="0"
                                    top="0"
                                    type="white"
                                    width="100%"
                        />
            </div>
</div>,
          "className": "plugin-body",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="plugin-body-main-col"
>
                <LoadingContent
                                bottom="0"
                                className=""
                                height="9em"
                                left="0"
                                right="0"
                                top="0"
                                type="white"
                                width="100%"
                />
                <LoadingContent
                                bottom="0"
                                className=""
                                height="9em"
                                left="0"
                                right="0"
                                top="1em"
                                type="white"
                                width="100%"
                />
</div>,
              <div
                className="plugin-body-side-col"
>
                <LoadingContent
                                bottom="0"
                                className=""
                                height="9em"
                                left="0"
                                right="0"
                                top="0"
                                type="white"
                                width="100%"
                />
</div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <LoadingContent
                    bottom="0"
                    className=""
                    height="9em"
                    left="0"
                    right="0"
                    top="0"
                    type="white"
                    width="100%"
/>,
                  <LoadingContent
                    bottom="0"
                    className=""
                    height="9em"
                    left="0"
                    right="0"
                    top="1em"
                    type="white"
                    width="100%"
/>,
                ],
                "className": "plugin-body-main-col",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "bottom": "0",
                    "className": "",
                    "height": "9em",
                    "left": "0",
                    "right": "0",
                    "top": "0",
                    "type": "white",
                    "width": "100%",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "bottom": "0",
                    "className": "",
                    "height": "9em",
                    "left": "0",
                    "right": "0",
                    "top": "1em",
                    "type": "white",
                    "width": "100%",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <LoadingContent
                  bottom="0"
                  className=""
                  height="9em"
                  left="0"
                  right="0"
                  top="0"
                  type="white"
                  width="100%"
/>,
                "className": "plugin-body-side-col",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "bottom": "0",
                  "className": "",
                  "height": "9em",
                  "left": "0",
                  "right": "0",
                  "top": "0",
                  "type": "white",
                  "width": "100%",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="plugin-header"
>
            <div
                        className="row no-flex"
            >
                        <LoadingContent
                                    bottom="0"
                                    className=""
                                    height="1.5em"
                                    left="0"
                                    right="0"
                                    top="0"
                                    type=""
                                    width="8em"
                        />
                        <LoadingContent
                                    bottom="0"
                                    className=""
                                    height="2.3em"
                                    left="0"
                                    right="0"
                                    top="1em"
                                    type=""
                                    width="12em"
                        />
                        <LoadingContent
                                    bottom="0"
                                    className=""
                                    height="1em"
                                    left="0"
                                    right="0"
                                    top="0.8em"
                                    type=""
                                    width="13em"
                        />
                        <LoadingContent
                                    bottom="0"
                                    className="loading-plugin-tag"
                                    height="1.5em"
                                    left="0"
                                    right="0"
                                    top="0.5em"
                                    type=""
                                    width="4em"
                        />
                        <LoadingContent
                                    bottom="0"
                                    className="loading-plugin-tag"
                                    height="1.5em"
                                    left="0.5em"
                                    right="0"
                                    top="1em"
                                    type=""
                                    width="14em"
                        />
            </div>
</div>,
          <div
            className="plugin-body"
>
            <div
                        className="row"
            >
                        <div
                                    className="plugin-body-main-col"
                        >
                                    <LoadingContent
                                                bottom="0"
                                                className=""
                                                height="9em"
                                                left="0"
                                                right="0"
                                                top="0"
                                                type="white"
                                                width="100%"
                                    />
                                    <LoadingContent
                                                bottom="0"
                                                className=""
                                                height="9em"
                                                left="0"
                                                right="0"
                                                top="1em"
                                                type="white"
                                                width="100%"
                                    />
                        </div>
                        <div
                                    className="plugin-body-side-col"
                        >
                                    <LoadingContent
                                                bottom="0"
                                                className=""
                                                height="9em"
                                                left="0"
                                                right="0"
                                                top="0"
                                                type="white"
                                                width="100%"
                                    />
                        </div>
            </div>
</div>,
        ],
        "className": "loading-plugin-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row no-flex"
>
              <LoadingContent
                            bottom="0"
                            className=""
                            height="1.5em"
                            left="0"
                            right="0"
                            top="0"
                            type=""
                            width="8em"
              />
              <LoadingContent
                            bottom="0"
                            className=""
                            height="2.3em"
                            left="0"
                            right="0"
                            top="1em"
                            type=""
                            width="12em"
              />
              <LoadingContent
                            bottom="0"
                            className=""
                            height="1em"
                            left="0"
                            right="0"
                            top="0.8em"
                            type=""
                            width="13em"
              />
              <LoadingContent
                            bottom="0"
                            className="loading-plugin-tag"
                            height="1.5em"
                            left="0"
                            right="0"
                            top="0.5em"
                            type=""
                            width="4em"
              />
              <LoadingContent
                            bottom="0"
                            className="loading-plugin-tag"
                            height="1.5em"
                            left="0.5em"
                            right="0"
                            top="1em"
                            type=""
                            width="14em"
              />
</div>,
            "className": "plugin-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <LoadingContent
                  bottom="0"
                  className=""
                  height="1.5em"
                  left="0"
                  right="0"
                  top="0"
                  type=""
                  width="8em"
/>,
                <LoadingContent
                  bottom="0"
                  className=""
                  height="2.3em"
                  left="0"
                  right="0"
                  top="1em"
                  type=""
                  width="12em"
/>,
                <LoadingContent
                  bottom="0"
                  className=""
                  height="1em"
                  left="0"
                  right="0"
                  top="0.8em"
                  type=""
                  width="13em"
/>,
                <LoadingContent
                  bottom="0"
                  className="loading-plugin-tag"
                  height="1.5em"
                  left="0"
                  right="0"
                  top="0.5em"
                  type=""
                  width="4em"
/>,
                <LoadingContent
                  bottom="0"
                  className="loading-plugin-tag"
                  height="1.5em"
                  left="0.5em"
                  right="0"
                  top="1em"
                  type=""
                  width="14em"
/>,
              ],
              "className": "row no-flex",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "bottom": "0",
                  "className": "",
                  "height": "1.5em",
                  "left": "0",
                  "right": "0",
                  "top": "0",
                  "type": "",
                  "width": "8em",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "bottom": "0",
                  "className": "",
                  "height": "2.3em",
                  "left": "0",
                  "right": "0",
                  "top": "1em",
                  "type": "",
                  "width": "12em",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "bottom": "0",
                  "className": "",
                  "height": "1em",
                  "left": "0",
                  "right": "0",
                  "top": "0.8em",
                  "type": "",
                  "width": "13em",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "bottom": "0",
                  "className": "loading-plugin-tag",
                  "height": "1.5em",
                  "left": "0",
                  "right": "0",
                  "top": "0.5em",
                  "type": "",
                  "width": "4em",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "bottom": "0",
                  "className": "loading-plugin-tag",
                  "height": "1.5em",
                  "left": "0.5em",
                  "right": "0",
                  "top": "1em",
                  "type": "",
                  "width": "14em",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
>
              <div
                            className="plugin-body-main-col"
              >
                            <LoadingContent
                                          bottom="0"
                                          className=""
                                          height="9em"
                                          left="0"
                                          right="0"
                                          top="0"
                                          type="white"
                                          width="100%"
                            />
                            <LoadingContent
                                          bottom="0"
                                          className=""
                                          height="9em"
                                          left="0"
                                          right="0"
                                          top="1em"
                                          type="white"
                                          width="100%"
                            />
              </div>
              <div
                            className="plugin-body-side-col"
              >
                            <LoadingContent
                                          bottom="0"
                                          className=""
                                          height="9em"
                                          left="0"
                                          right="0"
                                          top="0"
                                          type="white"
                                          width="100%"
                            />
              </div>
</div>,
            "className": "plugin-body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="plugin-body-main-col"
>
                  <LoadingContent
                                    bottom="0"
                                    className=""
                                    height="9em"
                                    left="0"
                                    right="0"
                                    top="0"
                                    type="white"
                                    width="100%"
                  />
                  <LoadingContent
                                    bottom="0"
                                    className=""
                                    height="9em"
                                    left="0"
                                    right="0"
                                    top="1em"
                                    type="white"
                                    width="100%"
                  />
</div>,
                <div
                  className="plugin-body-side-col"
>
                  <LoadingContent
                                    bottom="0"
                                    className=""
                                    height="9em"
                                    left="0"
                                    right="0"
                                    top="0"
                                    type="white"
                                    width="100%"
                  />
</div>,
              ],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <LoadingContent
                      bottom="0"
                      className=""
                      height="9em"
                      left="0"
                      right="0"
                      top="0"
                      type="white"
                      width="100%"
/>,
                    <LoadingContent
                      bottom="0"
                      className=""
                      height="9em"
                      left="0"
                      right="0"
                      top="1em"
                      type="white"
                      width="100%"
/>,
                  ],
                  "className": "plugin-body-main-col",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "bottom": "0",
                      "className": "",
                      "height": "9em",
                      "left": "0",
                      "right": "0",
                      "top": "0",
                      "type": "white",
                      "width": "100%",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "bottom": "0",
                      "className": "",
                      "height": "9em",
                      "left": "0",
                      "right": "0",
                      "top": "1em",
                      "type": "white",
                      "width": "100%",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <LoadingContent
                    bottom="0"
                    className=""
                    height="9em"
                    left="0"
                    right="0"
                    top="0"
                    type="white"
                    width="100%"
/>,
                  "className": "plugin-body-side-col",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "bottom": "0",
                    "className": "",
                    "height": "9em",
                    "left": "0",
                    "right": "0",
                    "top": "0",
                    "type": "white",
                    "width": "100%",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
